core/src/main/java/org/infinispan/CacheStream.java

```diff
@@ -241,13 +248,30 @@
 
    /**
     * {@inheritDoc}
-    * <p>This method has special usage when used with a distributed cache backing this set.  This operation will act
-    * as an intermediate iterator operation requiring data be brought locally for proper behavior.  This is
-    * described in more detail in the {@link CacheStream} documentation</p>
+    * <p>This operation will be invoked both remotely and locally when used with a distributed cache backing this stream.
+    * This operation will act as an intermediate iterator operation requiring data be brought locally for proper
+    * behavior.  This is described in more detail in the {@link CacheStream} documentation</p>
     * <p>This intermediate iterator operation will be performed locally and remotely requiring possibly a subset of
     * all elements to be in memory</p>
+    * <p>Any subsequent intermediate operations and the terminal operation are then performed locally.</p>
     * @return the new stream
     */
    @Override
    Stream<R> distinct();
+
+   /**
+    * {@inheritDoc}
+    * <p>Note when using a distributed backing cache for this stream the collector must be marshallable.  This
+    * prevents the usage of {@link java.util.stream.Collectors} class.  However you can use the
+    * {@link org.infinispan.stream.CacheCollectors} static factory methods to create a serializable wrapper, which then
```

Comment:
This is a great enhancement. With this in place, it's easier than ever to do distributed collects, and so I've added this method to Traversable for ease of use.